src/vmm/src/vstate/vcpu/mod.rs

```diff
@@ -866,21 +868,28 @@ mod tests {
     fn load_good_kernel(vm_memory: &GuestMemoryMmap) -> GuestAddress {
         use std::{fs::File, path::PathBuf};
 
-        let path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
-        let parent = path.parent().unwrap();
+        let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
 
         #[cfg(target_arch = "x86_64")]
-        let kernel_path: PathBuf = [parent.to_str().unwrap(), "kernel/src/loader/test_elf.bin"]
-            .iter()
-            .collect();
+        path.push("../../tests/resources/test_elf.bin");
```

Comment:
There is already `test_elf` and `test_pe` in `vmm/src/utilities/mock_resources`. Please try to use those since they are already in the same crate as this test. If all is good, we will be able to delete the once you relocated in `tests/resources`.